File: plugin-core/src/main/java/com/github/catatafishen/agentbridge/settings/ToolPermissionsConfigurable.java
package com.github.catatafishen.agentbridge.settings;

import com.github.catatafishen.agentbridge.services.ActiveAgentManager;
import com.github.catatafishen.agentbridge.services.ToolRegistry;
import com.github.catatafishen.agentbridge.ui.PermissionsPanel;
import com.intellij.openapi.options.Configurable;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

/**
 * Settings page wrapping the Tool Permissions panel.
 * Appears under Settings > Tools > AgentBridge > Tool Permissions.
 */
public final class ToolPermissionsConfigurable implements Configurable {

    private final Project project;
    private PermissionsPanel permissionsPanel;

    public ToolPermissionsConfigurable(@NotNull Project project) {
        this.project = project;
    }

    @Override
    public @Nls(capitalization = Nls.Capitalization.Title) String getDisplayName() {
        return "Tool Permissions";
    }

    @Override
    public @NotNull JComponent createComponent() {
        var settings = ActiveAgentManager.getInstance(project).getSettings();
        permissionsPanel = new PermissionsPanel(settings, ToolRegistry.getInstance(project),
            McpServerSettings.getInstance(project));
        return permissionsPanel.getComponent();
    }

    @Override
    public boolean isModified() {
        return permissionsPanel != null && permissionsPanel.isModified();
    }

    @Override
    public void apply() {
        if (permissionsPanel != null) {
            permissionsPanel.save();
        }
    }

    @Override
    public void reset() {
        if (permissionsPanel != null) {
            permissionsPanel.reload();
        }
    }

    @Override
    public void disposeUIResources() {
        permissionsPanel = null;
    }
}

File: plugin-core/src/main/java/com/github/catatafishen/agentbridge/settings/ToolPermissionsConfigurable.kt
package com.github.catatafishen.agentbridge.settings

import com.github.catatafishen.agentbridge.services.ActiveAgentManager
import com.github.catatafishen.agentbridge.services.ToolRegistry
import com.github.catatafishen.agentbridge.ui.PermissionsPanel
import com.intellij.openapi.options.SearchableConfigurable
import com.intellij.openapi.project.Project
import javax.swing.JComponent

class ToolPermissionsConfigurable(private val project: Project) : SearchableConfigurable {

    private var permissionsPanel: PermissionsPanel? = null

    override fun getId(): String = "com.github.catatafishen.agentbridge.tool-permissions"
    override fun getDisplayName(): String = "Tool Permissions"

    override fun createComponent(): JComponent {
        val settings = ActiveAgentManager.getInstance(project).settings
        val panel = PermissionsPanel(
            settings,
            ToolRegistry.getInstance(project),
            McpServerSettings.getInstance(project)
        )
        permissionsPanel = panel
        return panel.component
    }

    override fun isModified(): Boolean = permissionsPanel?.isModified == true
    override fun apply() { permissionsPanel?.save() }
    override fun reset() { permissionsPanel?.reload() }
    override fun disposeUIResources() { permissionsPanel = null }
}
